tests/Common/DevBranchesTest.php

```diff
@@ -11,32 +11,66 @@
 
 class DevBranchesTest extends StorageApiTestCase
 {
-    private static $cleanupAfterClassTokenId;
-
-    private static $teardownClient;
-
-    /**
-     * @dataProvider provideValidClients
-     */
-    public function testCreateBranch(Client $providedClient)
+    public function setUp()
     {
-        $providedToken = $providedClient->verifyToken();
-        $branches = new DevBranches($providedClient);
+        parent::setUp();
 
-        // cleanup
-        $branchesList = $branches->listBranches();
-        $branchName = __CLASS__ . '\\' . $this->getName() . '\\' . $providedToken['id'];
-        $branchesCreatedByThisTestMethod = array_filter(
-            $branchesList,
-            function ($branch) use ($branchName) {
-                return strpos($branch['name'], $branchName) === 0;
+        // branches cleanup
+        $branches = new DevBranches($this->_client);
+        foreach ($branches->listBranches() as $branch) {
+            if ($branch['isDefault'] !== true) {
+                $branches->deleteBranch($branch['id']);
             }
-        );
-        foreach ($branchesCreatedByThisTestMethod as $branch) {
-            $branches->deleteBranch($branch['id']);
         }
+    }
+
+    public function testCannotDeleteDefaultBranch()
+    {
+        $branches = new DevBranches($this->_client);
+        $branchesList = $branches->listBranches();
+
+        $this->assertCount(1, $branchesList);
+        $branch = reset($branchesList);
+
+        $this->assertSame(true, $branch['isDefault']);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionCode(400);
+        $this->expectExceptionMessage('Cannot delete Main branch');
+
+        $branches->deleteBranch($branch['id']);
+
+        $this->assertCount(1, $branches->listBranches());
+    }
+
+    public function testAdminRoleBranchesPermissions()
```

Comment:
Tenhle test netestuje nic moc jiného než testBranchCreateAndDelete, ne? 

Testuje, že admin to může vytvořit a smazat. Akorát je tady navíc to, že může smazat branch guesta. A že ji guest může vytvořit. Mohlo by to být věechno v jednom. Až budem mít storage tak to bdue epicky zpomalovat - to vytváření a zahazování branchí...